api/package.json

```
@@ -26,6 +26,16 @@
       "default": "./build/src/experimental/index.js"
     }
   },
+  "typesVersions": {
+    "<4.9": {
```

I think you want all TS versions.

 ```suggestion
    "*": {
```

In higher versions TS takes exports over this, but only if you have it set up in a specific esm-first way. You still want to use typesversions for commonjs.